modules/system/defaults/dock.nix

```diff
@@ -13,27 +13,79 @@ with lib;
       '';
     };
 
-    system.defaults.dock.orientation = mkOption {
-      type = types.nullOr (types.enum [ "bottom" "left" "right" ]);
+    system.defaults.dock.autohide-delay = mkOption {
+      type = types.nullOr types.attrs;
```

Comment:
I think this can be done in a nicer way with a custom type or an apply function. Currently this would require including the correct type when setting a value.

Reply:
Got pointers to docs/example somewhere, so I can try to address it?

Reply:
Something like this should allow using `isType "float" x` for the values.

```nix
let
  float = mkOptionType {
    name = "float";
    description = "float";
    check = x: isInt x || (isString x && builtins.match "[0-9]*\.[0-9]+" x != null);
    merge = loc: defs: { _type = "float"; value = mergeEqualOption loc defs; };
  };
in
  # ...
```